### PR TITLE
remove 404 page from sitemap 

### LINKED ISSUE #4: do not include the 404 error page

### PR DESCRIPTION
**Pull request by SOHELAHMED7**
fix #4

merged PR  #7, #10, #11 

after reviewing this PR there will no need to review PR of issues #7, #10, #11 

merge conflict solved

### CODE REVIEW

File: src/Module.php [hunk 1]
```diff
@@ -16,4 +16,12 @@ final class Module extends \luya\base\Module implements CoreModuleInterface
      * visible pages are included by default. In order to include hidden pages into the sitemap enable $withHidden.
      */
     public $withHidden = false;
+
+    /**
+     * @var boolean Whether the sitemap URLs should be encoded or not. Its highly recommend to enable this property as otherwise
+     * the Sitemap file could have errors and therfore won't work.
+     * @since 0.X.0
```
Comment (nadar) [inline src/Module.php:23]: @SOHELAHMED7 @cebe I was not sure which version will be released, therefore you should change this before or afterwards :-)

File: src/controllers/SitemapController.php [hunk 2]
```diff
@@ -61,27 +63,93 @@ private function buildSitemapfile($sitemapFile)
...
             if (!$this->module->withHidden) {
                 $query->andWhere(['is_hidden' => false]);
             }
-            
+
             foreach ($query->each() as $nav) {
                 /** @var Nav $nav */
 
+                if ((Config::findOne([
+                    'name' => 'httpExceptionNavId',
```
Comment (nadar) [inline src/controllers/SitemapController.php]: You could use this constant in favor of the variable string:
https://github.com/luyadev/luya-module-cms/blob/248c1803b246a40bbae482ff6b409720dc469211/src/models/Config.php#L20

(like on line#145)

Comment (SOHELAHMED7) [reply src/controllers/SitemapController.php]: fixed by 3e95b6ad64ae85271faf6e40c0e348bbda8dcfc5

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**